src/main/java/org/folio/circulation/resources/RequestCollectionResource.java

```diff
@@ -45,83 +45,71 @@ void create(RoutingContext routingContext) {
-    final RequestRepository requestRepository = RequestRepository.using(clients);
     final UserRepository userRepository = new UserRepository(clients);
     final LoanRepository loanRepository = new LoanRepository(clients);
-    final LoanPolicyRepository loanPolicyRepository = new LoanPolicyRepository(clients);
-    final RequestPolicyRepository requestPolicyRepository = new RequestPolicyRepository(clients);
-    final UpdateItem updateItem = new UpdateItem(clients);
-    final UpdateLoanActionHistory updateLoanActionHistory = new UpdateLoanActionHistory(clients);
-    final UpdateLoan updateLoan = new UpdateLoan(clients, loanRepository, loanPolicyRepository);
-    final ServicePointPickupLocationValidator servicePointPickupLocationValidator
-        = new ServicePointPickupLocationValidator();
-
-    final ProxyRelationshipValidator proxyRelationshipValidator =
-      createProxyRelationshipValidator(representation, clients);
 
     final CreateRequestService createRequestService = new CreateRequestService(
-      requestRepository, updateItem, updateLoanActionHistory, updateLoan, requestPolicyRepository);
-
-    final RequestRepresentation requestRepresentation = new RequestRepresentation();
-
-    final RequestFromRepresentationService requestFromRepresentationService
-      = new RequestFromRepresentationService(itemRepository, requestQueueRepository,
-          userRepository, loanRepository, servicePointRepository,
-          proxyRelationshipValidator, servicePointPickupLocationValidator);
+      RequestRepository.using(clients),
+      new UpdateItem(clients),
+      new UpdateLoanActionHistory(clients),
+      new UpdateLoan(clients, loanRepository, new LoanPolicyRepository(clients)),
+      new RequestPolicyRepository(clients)
+    );
+
+    final RequestFromRepresentationService requestFromRepresentationService =
+      new RequestFromRepresentationService(
+        new ItemRepository(clients, true, false),
+        RequestQueueRepository.using(clients),
+        userRepository,
+        loanRepository,
+        new ServicePointRepository(clients),
+        createProxyRelationshipValidator(representation, clients),
+        new ServicePointPickupLocationValidator()
+      );
 
     requestFromRepresentationService.getRequestFrom(representation)
       .thenComposeAsync(r -> r.after(createRequestService::createRequest))
       .thenApply(r -> r.map(RequestAndRelatedRecords::getRequest))
-      .thenApply(r -> r.map(requestRepresentation::extendedRepresentation))
+      .thenApply(r -> r.map(new RequestRepresentation()::extendedRepresentation))
       .thenApply(CreatedJsonResponseResult::from)
       .thenAccept(result -> result.writeTo(routingContext.response()));
   }
 
   void replace(RoutingContext routingContext) {
-    final WebContext context = new WebContext(routingContext);
 
     JsonObject representation = routingContext.getBodyAsJson();
+    write(representation, "id", getRequestId(routingContext));
```

Comment:
Can you help me understand why this was added?

I think it likely makes sense to add it, I just don't know what led to it being added during this change?

Reply:
Well It's because I moved this line and grouped representation handling

Reply:
Ok, I didn't spot the removed line :-/